azurerm/internal/services/cosmos/cosmosdb_account_resource.go

```diff
@@ -934,8 +960,9 @@ func resourceAzureRMCosmosDBAccountGeoLocationHash(v interface{}) int {
 	if m, ok := v.(map[string]interface{}); ok {
 		location := azure.NormalizeLocation(m["location"].(string))
 		priority := int32(m["failover_priority"].(int))
+		isZoneRedundant := m["is_zone_redundant"].(bool)
 
-		buf.WriteString(fmt.Sprintf("%s-%d", location, priority))
+		buf.WriteString(fmt.Sprintf("%s-%d-%v", location, priority, isZoneRedundant))
```

Comment:
I decide to revert this change, since we could not have two items with the same location and priority, but one of them has `true` for `zone_redundant` and another is `false`. Therefore I think there is no point to modify this hash function.